### DISCUSSION TITLE
How to compile

### DISCUSSION THREAD

**Discussion by notliterallyarobin** [otya128/winevdm#1454] [Q&A]
Hi,

I've installed Visual Studio 2017 and CMake but am unable to compile OTVDM with either method.

What needs to be edited within PropertySheet.props and/or what other dependencies are required to compile winevdm via VS2017 or CMake?

Thanks

**Comment by cracyc** [comment]
From PropertySheet.props 
```
    <!-- AsmPath should point the directory containing as.exe(binutils), bison.exe and flex.exe -->
    <AsmPath>C:\msys32\usr\bin\</AsmPath>
```
so you need as, bison and flex for the msvc compile.

**Reply by notliterallyarobin** [reply]
Thanks. I've installed Visual Studio 2017 (along with Windows SDK version 10.0.17134.0 as that seemed to be another dependency) and the 32-bit version of MSYS2 from [here](https://www.msys2.org/docs/windows_support/). To try and replicate the setup I've obtained bison.exe from [here](https://gnuwin32.sourceforge.net/packages/bison.htm), flex.exe from [here](https://gnuwin32.sourceforge.net/packages/flex.htm) and as.exe from an install of mingw32. All three have been placed in C:\msys32\usr\bin. Is MSYS2 actually required or is it just where you happen to have your as.exe, bison.exe and flex.exe binaries?

It does now build an otvdm.exe and otvdmw.exe, both of which appear to work, although I do still get the following errors:
![image](https://github.com/user-attachments/assets/42fab986-6727-4f5e-a262-39b08952a003)

Full build output is here: [build_output.txt](https://github.com/user-attachments/files/17458648/build_output.txt)

I'm also assuming this isn't the same version of Bison that you're using as building with CMake fails when trying to identify the version as bison --version doesn't give any output: [cmake_output.txt](https://github.com/user-attachments/files/17458821/cmake_output.txt)

**Reply by otya128** [reply]
That flex is based on a version that is nearly 30 years old and cannot be used.

```
5>Performing Custom Build Tools
5>"ppl.l", line 124: unrecognized '%' directive
5>"ppl.l", line 125: bad character: #
5>"ppl.l", line 125: unknown error processing section 1
5>"ppl.l", line 125: unknown error processing section 1
5>"ppl.l", line 125: bad character: "
5>"ppl.l", line 125: unknown error processing section 1
5>"ppl.l", line 125: bad character: .
5>"ppl.l", line 125: unknown error processing section 1
5>"ppl.l", line 125: bad character: "
5>"ppl.l", line 126: bad character: #
5>"ppl.l", line 126: unknown error processing section 1
5>"ppl.l", line 126: bad character: "
5>"ppl.l", line 126: unknown error processing section 1
5>"ppl.l", line 126: bad character: /
5>"ppl.l", line 126: unknown error processing section 1
5>"ppl.l", line 126: bad character: .
5>"ppl.l", line 126: unknown error processing section 1
5>"ppl.l", line 126: bad character: "
5>"ppl.l", line 127: bad character: }
5>Performing Custom Build Tools
5>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(209,5): error MSB6006: "cmd.exe" exited with code -1073741515.
```

Execute the following command on the MSYS2 shell to install bison and flex for MSYS2.

```sh
pacman -S flex bison
```
